cmd/microcloud/ask.go

```diff
@@ -914,6 +914,13 @@ func (c *initConfig) askRemotePool(sh *service.Handler) error {
 	return nil
 }
 
+// ipWithIface is a helper struct to store an IP address and
+// its corresponding network interface.
+type ipWithIface struct {
+	ip        net.IP
+	ifaceName string
+}
+
```

Comment:
Instead of maintaining both this and several fields for each interface on `InitSystem`, why not just introduce a new type `Network` with 3 fields: `Interface`, `IP`, `Subnet`.

Then each of `OVNGeneveAddr` `MicroCephInternalSubnet`, and `MicroCephPublicSubnet` can be changed to `OVNGeneveNetwork`, `CephInternalNetwork`, `CephPublicNetwork` respectively.

And finally `validateSystems` you can check per-cluster member if any local interfaces clash, and then across all cluster members if the subnets clash.

Reply:
I agree